cmd/medius/images/verify.go

```diff
@@ -0,0 +1,252 @@
+			client, err := kvirtcli.GetKubevirtClient()
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			wg, errChan := spawnWorkers(options.VerifyImagesOptions.Workers, options.Focus, func(a api.Artifact) error {
+				return verifyArtifact(a, options, client)
+			})
+			wg.Wait()
+
+			select {
+			case <-errChan:
+				os.Exit(1)
+			default:
+				os.Exit(0)
+			}
+		},
+	}
+	verifyCmd.Flags().IntVar(&options.VerifyImagesOptions.Workers, "workers", options.VerifyImagesOptions.Workers, "Number of parallel workers")
+	verifyCmd.Flags().StringVar(&options.VerifyImagesOptions.ClusterRegistry, "cluster-registry", "", "Registry to use inside the cluster")
+	verifyCmd.Flags().StringVar(&options.VerifyImagesOptions.Namespace, "namespace", options.VerifyImagesOptions.Namespace, "Namespace to run verify in")
+	verifyCmd.Flags().IntVar(&options.VerifyImagesOptions.Timeout, "timeout", options.VerifyImagesOptions.Timeout, "Maximum seconds to wait for VM to be running")
+	verifyCmd.Flags().AddGoFlagSet(kvirtcli.FlagSet())
+
+	return verifyCmd
+}
+
+func verifyArtifact(artifact api.Artifact, options *common.Options, client kvirtcli.KubevirtClient) (err error) {
+	log := common.Logger(artifact)
+
+	imgRefs, err := findImgRefs(artifact, options, log)
+	if err != nil {
+		log.WithError(err).Error("Failed to find containerdisks")
+		return err
+	}
+
+	if len(imgRefs) == 0 {
+		log.Infof("Found no containerdisks to verify")
+		return nil
+	}
+
+	vm, privateKey, err := createVM(artifact, options, imgRefs[0])
+	if err != nil {
+		log.WithError(err).Error("Failed to create VM object")
+		return err
+	}
+
+	vmClient := client.VirtualMachine(options.VerifyImagesOptions.Namespace)
+	log.Info("Creating VM")
+	if vm, err = vmClient.Create(vm); err != nil {
+		log.WithError(err).Error("Failed to create VM")
+		return err
+	}
+
+	defer func() {
+		if err = vmClient.Delete(vm.Name, &metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64(0)}); err != nil {
+			log.WithError(err).Error("Failed to delete VM")
+		}
+	}()
+
+	log.Info("Waiting for VM to be ready")
+	if err = waitVMReady(vm.Name, vmClient, options.VerifyImagesOptions.Timeout); err != nil {
+		log.WithError(err).Error("VM not ready")
+		return err
+	}
+
+	vmi, err := client.VirtualMachineInstance(options.VerifyImagesOptions.Namespace).Get(vm.Name, &metav1.GetOptions{})
+	if err != nil {
+		log.WithError(err).Error("Failed to get VMI")
+		return err
+	}
+
+	log.Info("Running tests on VMI")
+	for _, testFn := range artifact.Tests() {
+		if err = testFn(vmi, &api.ArtifactTestParams{Username: VerifyUsername, PrivateKey: privateKey}); err != nil {
+			log.WithError(err).Error("Failed to verify containerdisk")
+			return err
+		}
+	}
+
+	imgRefs = append(imgRefs, prepareTimestampTag(options.Registry, artifact.Metadata()))
+	if err = pushImages(imgRefs, options, log); err != nil {
+		log.WithError(err).Error("Failed to update containerdisks")
+		return err
+	}
+
+	return err
```

Comment:
Can you return `nil` here, to make it obvious that there was no error?

Reply:
Resolved